### ISSUE TITLE
[Safari] Annoying refresh on every search

### ISSUE THREAD

**Issue by vicmarto** [ZenithO-o/Fix-Google-Web-Search#2]
In Safari 17.5 on macOS Sonoma 14.5 (using Userscripts 4.4.5), the script causes an annoying refresh:

1) On every search, it first loads Google with the “All” tab.
2) It then quickly switches the tab to “Web” (which causes a fast refresh).

Is there any way to avoid this?

**Comment by isaacl** [comment]
tl;dr  no it's not possible to fix this.

----

The various browser search bars load a page with google general search each time, even if your previous search was in web category.

Potentially there is a cookie google would look at to determine utm parameter but that requires reverse engineering google search and is brittle.

Also you could potentially use a custom search engine with google but with the utm parameter added, but that will require extra typing each time you want to search from the box.

Blame Google for [making their search worse](https://www.wheresyoured.at/the-men-who-killed-google/).

**Comment by vicmarto** [comment]
I see. Thank you very much for your detailed reply.

Instead, using Brave (with a custom search engine), it is possible to jump directly to the “Web” tab, avoiding the "All” tab.

Very interesting reading your link, like all of us I too have noticed the gradual deterioration of Google results in these last years. Now I know the reason and the responsible... very sad everything....

I think the best thing to do will be to pass this patch called 'udm=14', and directly abandon the use of Google.

**Comment by isaacl** [comment]
@vicmarto  Glad you liked the link.  It's true: many platforms I rely on are in a sad state these days.

You may be interested in [Kagi](https://kagi.com/) search engine.  They have their own crawler, but also indirectly query google for your search (API), and then they filter/reorder google's own results before presenting to you.

I'm currently still using google but this is probably where I'd go if google gets worse.

**Comment by niko-vv** [comment]
For me, I think adding // @run-at document-start between line 7 and 8 solved this problem (on MacOS Chrome). I haven't really used it enough to see how well it works but appears to remove the refresh thing. On my Pixel 5 with ChromeXt, changing the setting for the userscript to Run At: Start fixed it also.

**Comment by isaacl** [comment]
@niko-vv That certainly can't fix the redirection but it might make it so fast that you don't notice anymore.  I'll send in a PR
